Model/Cart/Item/Builder.php

```diff
@@ -124,4 +125,30 @@ public function buildItemId(Item $item)
 
         return (string)$item->getProduct()->getId();
     }
+
+    /**
+     * Returns the sku id. If it is a configurable product,
+     * try to get the child item because the child item is the simple product
+     *
+     * @param Item $item the sales item model.
+     * @return string|null sku id
+     */
+    public function buildSkuId(Item $item)
+    {
+        if ($item->getProductType() === Configurable::getType()) {
+            $children = $item->getChildren();
+            //An item with bundle product and group product may have more than 1 child.
+            //But configurable product item should have max 1 child item.
+            //Here we check the size of children, return only if the size is 1
+            if (count($children) == 1
+                && array_key_exists(0, $children)
+            ) {
+                if ($children[0]->getProduct()) {
```

Comment:
Should we check that `$children[0]->getProduct()` returns `Magento\Catalog\Model\Product`?